Upgrade dependencies for Vagrant image

arjansh opened the pull request:
The files hadoop-2.7.1.tar.gz, apache-maven-3.3.3-bin.tar.gz, oozie-4.2.0.tar.gz and ext-2.2.zip are no longer available for download, so I upgrade the first three to newer versions which are available for download and I added a troubleshooting section to a new README.md describing that you have to download the ext-2.2.zip file yourself by finding it by yourself on the internet (which worked for me).

arjansh committed:
the files hadoop-2.7.1.tar.gz, apache-maven-3.3.3-bin.tar.gz, oozie-4.2.0.tar.gz and ext-2.2.zip are no longer available for download, so I upgrade the first three to newer versions which are available for download and I added a troubleshooting section to a new README.md describing that you have to download the ext-2.2.zip file yourself by finding it by yourself on the internet (which worked for me).

jhorcicka commented:
For Oozie, http://mirrors.rackhosting.com/apache/oozie/4.3.0/oozie-4.3.0.tar.gz link is used, but that gives 404 right now. Because of that "vagrant up" prints many complains about oozie during the startup. 

I belive http://mirrors.rackhosting.com/apache/oozie/4.3.1/oozie-4.3.1.tar.gz should be used instead.

Commit:
Updating oozie version and using a different location to get ext-2.2.zip from.

jhorcicka commented:
Now I get "hadoop-2.7.4/sbin/start-yarn.sh: No such file or directory". 

For "http://mirrors.rackhosting.com/apache/hadoop/common/hadoop-2.7.4/hadoop-2.7.4.tar.gz", I get 404. 

```
    default: Downloaded: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-core/1.0-alpha-10/doxia-core-1.0-alpha-10.jar (54 KB at 2667.7 KB/sec)
    default: [INFO] 
    default: [INFO] --- maven-assembly-plugin:2.2.1:single (default-cli) @ oozie-main ---
    default: [INFO] Reading assembly descriptor: src/main/assemblies/empty.xml
    default: [INFO] ------------------------------------------------------------------------
    default: [INFO] Reactor Summary:
    default: [INFO] 
    default: [INFO] Apache Oozie Hadoop Utils hadoop-2-4.3.1 ........... SUCCESS [02:26 min]
    default: [INFO] Apache Oozie Hadoop Distcp hadoop-2-4.3.1 .......... SUCCESS [  0.638 s]
    default: [INFO] Apache Oozie Hadoop Auth hadoop-2-4.3.1 Test ....... SUCCESS [  0.356 s]
    default: [INFO] Apache Oozie Hadoop Libs ........................... SUCCESS [  0.045 s]
    default: [INFO] Apache Oozie Client ................................ SUCCESS [01:00 min]
    default: [INFO] Apache Oozie Share Lib Oozie ....................... SUCCESS [ 50.607 s]
    default: [INFO] Apache Oozie Share Lib HCatalog .................... SUCCESS [ 10.002 s]
    default: [INFO] Apache Oozie Share Lib Distcp ...................... SUCCESS [  1.241 s]
    default: [INFO] Apache Oozie Core .................................. SUCCESS [01:30 min]
    default: [INFO] Apache Oozie Share Lib Streaming ................... SUCCESS [  6.640 s]
    default: [INFO] Apache Oozie Share Lib Pig ......................... SUCCESS [ 14.975 s]
    default: [INFO] Apache Oozie Share Lib Hive ........................ SUCCESS [ 12.024 s]
    default: [INFO] Apache Oozie Share Lib Hive 2 ...................... SUCCESS [  7.405 s]
    default: [INFO] Apache Oozie Share Lib Sqoop ....................... SUCCESS [  4.649 s]
    default: [INFO] Apache Oozie Examples .............................. SUCCESS [ 14.192 s]
    default: [INFO] Apache Oozie Share Lib Spark ....................... SUCCESS [ 23.869 s]
    default: [INFO] Apache Oozie Share Lib ............................. SUCCESS [ 18.066 s]
    default: [INFO] Apache Oozie Docs .................................. SUCCESS [ 24.556 s]
    default: [INFO] Apache Oozie WebApp ................................ SUCCESS [01:49 min]
    default: [INFO] Apache Oozie Tools ................................. SUCCESS [  4.938 s]
    default: [INFO] Apache Oozie MiniOozie ............................. SUCCESS [  2.364 s]
    default: [INFO] Apache Oozie Distro ................................ SUCCESS [ 27.044 s]
    default: [INFO] Apache Oozie ZooKeeper Security Tests .............. SUCCESS [  8.720 s]
    default: [INFO] Apache Oozie Main .................................. SUCCESS [  5.136 s]
    default: [INFO] ------------------------------------------------------------------------
    default: [INFO] BUILD SUCCESS
    default: [INFO] ------------------------------------------------------------------------
    default: [INFO] Total time: 12:14 min
    default: [INFO] Finished at: 2018-06-19T12:19:18+00:00
    default: [INFO] Final Memory: 536M/957M
    default: [INFO] ------------------------------------------------------------------------
    default: Oozie distro created, DATE[2018.06.19-12:07:02GMT] VC-REV[unavailable], available at [/home/vagrant/oozie-4.3.1/distro/target]
    default: Configuring Oozie
    default: Overwriting conf/hadoop-conf contents for Oozie
==> default: Running provisioner: shell...
    default: Running: inline script
    default: Starting DFS
    default: /tmp/vagrant-shell: line 2: hadoop-2.7.4/sbin/start-dfs.sh: No such file or directory
    default: Starting HBase
    default: bigdatavm: Warning: Permanently added 'bigdatavm,192.168.33.10' (ECDSA) to the list of known hosts.
    default: bigdatavm: starting zookeeper, logging to /home/vagrant/hbase-1.1.1/bin/../logs/hbase-vagrant-zookeeper-bigdatavm.out
    default: starting master, logging to /home/vagrant/hbase-1.1.1/bin/../logs/hbase-vagrant-master-bigdatavm.out
    default: Java HotSpot(TM) 64-Bit Server VM warning: ignoring option PermSize=128m; support was removed in 8.0
    default: Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=128m; support was removed in 8.0
    default: starting regionserver, logging to /home/vagrant/hbase-1.1.1/bin/../logs/hbase-vagrant-1-regionserver-bigdatavm.out
    default: Starting YARN
    default: /tmp/vagrant-shell: line 6: hadoop-2.7.4/sbin/start-yarn.sh: No such file or directory
    default: Starting MR historyserver
    default: /tmp/vagrant-shell: line 8: hadoop-2.7.4/sbin/mr-jobhistory-daemon.sh: No such file or directory
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Commit:
Merge branch 'master' into feature/upgrade-bigdatavm-downloads

Commit:
Changes:
- Updated ubuntu version from trusty64 to xenial64
- allocate more memory for virtual box
- Install Java openjdk8
- Download hadoop-2.7.4 from different archive
- No longer override Oozie pom.xml
- Changed resource definitions for yarn

Commit:
Increased memory size.

jhorcicka commented:
First I had to change the Vagrant file like this: 

![01](https://user-images.githubusercontent.com/13213915/60596662-c1ebca00-9da9-11e9-801f-f819cd018807.png)

![02](https://user-images.githubusercontent.com/13213915/60596663-c1ebca00-9da9-11e9-9715-bc2a706379ee.png)

Then I get the following error: 
C:\work\vagrant-vms\bigdatavm>vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'ubuntu/trusty64' version '20190429.0.1' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 8020 (guest) => 8020 (host) (adapter 1)
    default: 9000 (guest) => 9000 (host) (adapter 1)
    default: 50070 (guest) => 50070 (host) (adapter 1)
    default: 50090 (guest) => 50090 (host) (adapter 1)
    default: 50075 (guest) => 50075 (host) (adapter 1)
    default: 50010 (guest) => 50010 (host) (adapter 1)
    default: 16010 (guest) => 16010 (host) (adapter 1)
    default: 16030 (guest) => 16030 (host) (adapter 1)
    default: 16000 (guest) => 16000 (host) (adapter 1)
    default: 16020 (guest) => 16020 (host) (adapter 1)
    default: 2181 (guest) => 2181 (host) (adapter 1)
    default: 8032 (guest) => 8032 (host) (adapter 1)
    default: 8088 (guest) => 8088 (host) (adapter 1)
    default: 50060 (guest) => 50060 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 4.3.40
    default: VirtualBox Version: 6.0
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
==> default: Mounting shared folders...
    default: /vagrant => C:/work/vagrant-vms/bigdatavm
==> default: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> default: flag to force provisioning. Provisioners marked to run always will still run.
==> default: Running provisioner: shell...
    default: Running: inline script
    default: Starting DFS
    default: /tmp/vagrant-shell: line 2: hadoop-2.7.4/sbin/start-dfs.sh: No such file or directory
    default: Starting HBase
    default: /home/vagrant/hbase-1.1.1/bin/hbase: line 402: /usr/lib/jvm/java-8-oracle/jre/bin/java: No such file or directory
    default: /home/vagrant/hbase-1.1.1/bin/hbase: line 402: /usr/lib/jvm/java-8-oracle/jre/bin/java: No such file or directory
    default: starting master, logging to /home/vagrant/hbase-1.1.1/bin/../logs/hbase-vagrant-master-bigdatavm.out
    default: /home/vagrant/hbase-1.1.1/bin/../bin/hbase: line 402: /usr/lib/jvm/java-8-oracle/jre/bin/java: No such file or directory
    default: starting regionserver, logging to /home/vagrant/hbase-1.1.1/bin/../logs/hbase-vagrant-1-regionserver-bigdatavm.out
    default: /home/vagrant/hbase-1.1.1/bin/../bin/hbase: line 402: /usr/lib/jvm/java-8-oracle/jre/bin/java: No such file or directory
    default: Starting YARN
    default: /tmp/vagrant-shell: line 6: hadoop-2.7.4/sbin/start-yarn.sh: No such file or directory
    default: Starting MR historyserver
    default: /tmp/vagrant-shell: line 8: hadoop-2.7.4/sbin/mr-jobhistory-daemon.sh: No such file or directory
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

jhorcicka commented:
This was reviewed&tested by Gerard Dellemann. I am merging this on his behalf.
